stablehlo/compatibility/dialect/VersionedStablehloAttrs.td

```diff
@@ -0,0 +1,174 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_COMPATIBILITY_VERSIONEDHLO_ATTRS
+#define STABLEHLO_COMPATIBILITY_VERSIONEDHLO_ATTRS
+
+include "stablehlo/compatibility/dialect/VersionedStablehloBase.td"
+include "stablehlo/compatibility/dialect/VersionedStablehloEnums.td"
+
+include "mlir/IR/AttrTypeBase.td"
+
+//===----------------------------------------------------------------------===//
+// Attributes
+//===----------------------------------------------------------------------===//
+
+def VersionedHLO_Dim : ArrayRefParameter<"int64_t", "Dimension"> {
+  let parser = "mlir::hlo::parseIntArray($_parser)";
+  let printer = "mlir::hlo::printIntArray($_printer, $_self)";
+}
+
+def VersionedHLO_ScatterDimensionNumbers : AttrDef<VersionedHLO_Dialect, "ScatterDimensionNumbers"> {
+  let cppNamespace = "::mlir::stablehlo";
+  let mnemonic = "scatter";
+  let summary = "Attribute that models the dimension information for scatter";
+  let parameters = (ins
+      VersionedHLO_Dim:$updateWindowDims,
+      VersionedHLO_Dim:$insertedWindowDims,
+      VersionedHLO_Dim:$scatterDimsToOperandDims,
+      "int64_t":$indexVectorDim
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+def VersionedHLO_GatherDimensionNumbers : AttrDef<VersionedHLO_Dialect, "GatherDimensionNumbers"> {
+  let cppNamespace = "::mlir::stablehlo";
+  let mnemonic = "gather";
+  let parameters = (ins
+      VersionedHLO_Dim:$offsetDims,
+      VersionedHLO_Dim:$collapsedSliceDims,
+      VersionedHLO_Dim:$startIndexMap,
+      "int64_t":$indexVectorDim
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+def VersionedHLO_DotDimensionNumbers : AttrDef<VersionedHLO_Dialect, "DotDimensionNumbers"> {
+  let cppNamespace = "::mlir::stablehlo";
+  let mnemonic = "dot";
+  let summary = "Attribute that models the dimension information for dot.";
+  let parameters = (ins
+      VersionedHLO_Dim:$lhsBatchingDimensions,
+      VersionedHLO_Dim:$rhsBatchingDimensions,
+      VersionedHLO_Dim:$lhsContractingDimensions,
+      VersionedHLO_Dim:$rhsContractingDimensions
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+def VersionedHLO_ArgResultAlias : AttrDef<VersionedHLO_Dialect, "ArgResultAlias"> {
+  let cppNamespace = "::mlir::stablehlo";
+  let mnemonic = "result_alias";
+  let parameters = (ins
+    VersionedHLO_Dim:$argTupleIndices,
+    "int64_t":$resultIndex,
+    VersionedHLO_Dim:$resultTupleIndices,
+    "bool":$isMustAlias
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
+}
+
+def OutputOperandAlias : AttrDef<VersionedHLO_Dialect, "OutputOperandAlias"> {
```

Comment:
Huh. This happened because it is called `OutputOperandAlias` in `StablehloAttrs.td` instead of `StableHLO_OutputOperandAlias`. I'll make that fix too.